### PR TITLE
Improvements to API AM intro doc in use cases directory

### PR DESCRIPTION
**Pull request by mystiberry-okta**
@tthompson-okta  This has everything but the images and the extra links to oauth.com. Am I headed in the right direction? And how can I include the images when the docs never discuss resource server (the team told me not to)?  Thanks!

### CODE REVIEW

File: _source/_use_cases/api_security/index.md [hunk 1]
```diff
@@ -45,8 +44,41 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
 * Enjoy the highest quality, always-available API Access Management. 
 * Let Okta do the work of consuming standards changes to provide more or better services.
+* End users or the target service own their data.
+* End users or the application owner specify the authorization policy.
+
+> Note: In some places we have implemented stricter requirements or behaviors for additional security. 
+ 
+## OAuth 2.0 and Other Okta Features
+
+Okta provides two open-source features that work with OAuth 2.0: a sign-in widget and a JavaScript SDK.
```
Comment (tthompson-okta) [inline _source/_use_cases/api_security/index.md]: They're both technically "javascript SDKs" and one wraps the other; it just depends on whether you want to own the UX and build your own login flow or whether you want the drop-in convenience of a widget.

File: _source/_use_cases/api_security/index.md [hunk 2]
```diff
@@ -45,8 +44,41 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+
+### Okta Auth SDK
+
+[The Okta Auth SDK](https://github.com/okta/okta-auth-js) is also open source. 
+It wraps Okta authentication and session management APIs in JavaScript callbacks for custom user experiences that require functionality
+beyond what the Sign-in Widget can provide.
+This SDK implements custom web messaging with OAuth 2.0 response mode (okta-post-message) for single page applications with a hidden iframe.
+
+## Specifications Implemented by Okta
+
```
Comment (tthompson-okta) [inline _source/_use_cases/api_security/index.md]: We should hyperlink to each of these specs as well

**Comment by tthompson-okta** [comment]
Yep this is on the right track. I left some clarification on the Jira as to why the diagrams are okay 👍

File: _source/_use_cases/api_security/index.md [hunk 3]
```diff
@@ -7,11 +7,17 @@ excerpt: Secure your APIs with Okta's implementation of the OAuth 2.0 standard.
...
 # Overview
 
-Use API Access Management, Okta's implementation of the OAuth 2.0 standard, to secure your APIs.
+Use API Access Management, Okta's implementation of the OAuth 2.0 protocol, to secure your APIs.
 API Access Management is integrated with Okta's implementation of OpenID Connect for authentication;
 OpenID Connect is also available separately. Similarly, Okta provides a client management API
 for onboarding, monitoring, and deprovisioning client apps.
 
+[OAuth 2.0](https://www.oauth.com) is an open standard that defines how to perform authentication securely. It decouples
+authorization policy decisions from enforcement. API Access Management is Okta's implementation an authorization service which leverages Okta's
```
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: Nit: ```implementation of an authorization..```

File: _source/_use_cases/api_security/index.md [hunk 4]
```diff
@@ -7,11 +7,17 @@ excerpt: Secure your APIs with Okta's implementation of the OAuth 2.0 standard.
...
+Universal directory, groups, policies, multifactor authentication (MFA) and other features.
```
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: Universal Directory (capitalize the D)

File: _source/_use_cases/api_security/index.md [hunk 5]
```diff
@@ -45,14 +46,59 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
 * Control complex business requirements with polices and rules. You control the ordering and relationships.
 * Enjoy the highest quality, always-available API Access Management. 
 * Let Okta do the work of consuming standards changes to provide more or better services.
+* End users or the target service own their data.
+* End users or the application owner specify the authorization policy.
+
+> Note: In some places we have implemented stricter requirements or behaviors for additional security. 
+ 
+## OAuth 2.0 and Other Okta Features
```
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: Nit: 'Other Okta Features' is very nebulous and doesn't tell the reader much. Instead, we are highlighting some key and unique platform offerings. Can we jazz up the wording somehow?

File: _source/_use_cases/api_security/index.md [hunk 6]
```diff
@@ -45,14 +46,59 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+
+Okta provides two open-source features that work with OAuth 2.0: a sign-in widget and a JavaScript SDK.
+
+### Okta Sign-in Widget
+
+[The Sign-in Widget](https://github.com/okta/okta-signin-widget) is Okta's open-source, embeddable widget for end-to-end sign-in flow with MFA and account recovery:
+
+* Returns an ID Token for an authenticated user 
+* Supports social authentication providers
```
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: Would be good to link to our social auth page so one can discover which social auth providers we support.

File: _source/_use_cases/api_security/index.md [hunk 7]
```diff
@@ -45,14 +46,59 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+* OAuth 2.0 Multiple Response Types (spec) 
+* OAuth 2.0 Form Post Response Mode (spec)
+* OAuth 2.0 Token Revocation (RFC 7009) 
+* OAuth 2.0 Token Introspection (RFC 7662) 
+* Proof Key for Code Exchange by OAuth Public Clients (RFC 7636)
 
 ## Putting the Pieces Together
 
 The following is a high-level look at the basic components of API Access Management. 
 We use the same terms as the OpenID Connect and OAuth 2.0 spec. For complete explanations, read those specs.
```
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: Do you want to link to the specs here? Or is it done elsewhere in this doc which is quickly discoverable?

File: _source/_use_cases/api_security/index.md [hunk 8]
```diff
@@ -45,14 +46,59 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
 
+### Resource Servers and Authentication Servers
+
+The [OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749#section-1.1) defines a resource server, owned
+and operated by the API owner, which consumes access tokens. 
+The spec also defines an authorization server, which Okta hosts as part of the API Access Management feature, which mints tokens.
```
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: which mints access tokens.

File: _source/_use_cases/api_security/index.md [hunk 9]
```diff
@@ -7,11 +7,17 @@ excerpt: Secure your APIs with Okta's implementation of the OAuth 2.0 standard.
 
 # Overview
 
-Use API Access Management, Okta's implementation of the OAuth 2.0 standard, to secure your APIs.
+Use API Access Management, Okta's implementation of the OAuth 2.0 protocol, to secure your APIs.
 API Access Management is integrated with Okta's implementation of OpenID Connect for authentication;
 OpenID Connect is also available separately. Similarly, Okta provides a client management API
 for onboarding, monitoring, and deprovisioning client apps.
 
+[OAuth 2.0](https://www.oauth.com) is an open standard that defines how to perform authentication securely. It decouples
```
Comment (yuliu-okta) [inline _source/_use_cases/api_security/index.md]: authentication -> authorization

File: _source/_use_cases/api_security/index.md [hunk 10]
```diff
@@ -45,14 +46,59 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
 The following is a high-level look at the basic components of API Access Management. 
 We use the same terms as the OpenID Connect and OAuth 2.0 spec. For complete explanations, read those specs.
 
+### Resource Servers and Authentication Servers
+
+The [OAuth 2.0 spec](https://tools.ietf.org/html/rfc6749#section-1.1) defines a resource server, owned
+and operated by the API owner, which consumes access tokens. 
+The spec also defines an authorization server, which Okta hosts as part of the API Access Management feature, which mints tokens.
+
+<img src="/assets/img/app-auth-server.png" alt="Resource Server" width="640px" />
```
Comment (yuliu-okta) [inline _source/_use_cases/api_security/index.md]: you mean the feature for private resource server? We have already killed it.

File: _source/_use_cases/api_security/index.md [hunk 11]
```diff
@@ -45,14 +46,59 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
 * Control complex business requirements with polices and rules. You control the ordering and relationships.
 * Enjoy the highest quality, always-available API Access Management.
```
Comment (tthompson-okta) [inline _source/_use_cases/api_security/index.md]: These bullets points sounds like they're reaching just a little bit; can we remove them?

> Enjoy the highest quality, always­ available API Access Management.
> End users or the target service own their data.
> End users or the application owner specify the authorization policy.

File: _source/_use_cases/api_security/index.md (hunk 1 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: I've cleaned this up, using your comments and existing content in the referenced pages: 

## OAuth 2.0 and Other Okta Features

Okta provides two open-source features that work with OAuth 2.0:
 
* [The Okta Sign-In Widget](/code/javascript/okta_sign-in_widget.html)gives you a customizable login experience
with the drop-in convenience of a JavaScript widget.

* [The Okta Auth SDK](/code/javascript/okta_auth_sdk.html) is a wrapper around the Sign-In Widget, and allows you
to own the user experience and build your own login flow.

File: _source/_use_cases/api_security/index.md (hunk 2 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

File: _source/_use_cases/api_security/index.md (hunk 3 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

File: _source/_use_cases/api_security/index.md (hunk 4 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

File: _source/_use_cases/api_security/index.md (hunk 6 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: I beefed up the text, how is this for a better title: 

Customizing Login Flows with Okta and OAuth 2.0

File: _source/_use_cases/api_security/index.md (hunk 7 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Links added!

File: _source/_use_cases/api_security/index.md (hunk 8 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Fixed.

File: _source/_use_cases/api_security/index.md (hunk 9 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Fixed.

File: _source/_use_cases/api_security/index.md (hunk 10 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: No. Trevor has given me some notes for how to clarify this.

File: _source/_use_cases/api_security/index.md (hunk 11 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

**Comment by mystiberry-okta** [comment]
Need to check Trevor's comments and then will send a new commit for review.

File: _source/_use_cases/api_security/index.md [hunk 12]
```diff
@@ -43,16 +44,93 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+Okta provides two types of authorization servers:
+
+* [Org Authorization Server](#org-authorization-server): built-in with no configuration and simple behaviors
+* [Resource-Domain Authorization Server](resource-domain-authorization-server): configurable and rich resource access policy
+
+#### Org Authorization Server
+
+Okta's Org Authorization Server requires no set up or configuration to mint ID Tokens and Access Tokens. 
+This authorization server supports OpenID Connect-only use cases.
+
```
Comment (richardmateosian-okta) [inline _source/_use_cases/api_security/index.md]: How about "supports only OpenID Connect use cases"?

File: _source/_use_cases/api_security/index.md [hunk 13]
```diff
@@ -43,16 +44,93 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+* Access policies are assignment based. Access is granted based on whether the user is assigned to the client app in Okta (directly or via groups).
+* Scopes in a request are either all granted, or none are. 
+* Audience (`aud`) for Access Tokens is always the Okta org:
+    * If an Access Token is requested, it's only good for Okta APIs. The token can't be used for custom or internal APIs.
+    * The Access Token format is opaque and shouldn't be parsed by the client. 
+* The Org Authorization Server is referenced via the root OAuth namespace: `/oauth2/v1/[authorize|token|introspect|revoke]`.
+
```
Comment (richardmateosian-okta) [inline _source/_use_cases/api_security/index.md]: How about "uses" rather than "is referenced via"

File: _source/_use_cases/api_security/index.md [hunk 14]
```diff
@@ -65,6 +143,9 @@ Both of these measures go a long way toward mitigating the impact of a security
 
 Okta helps you manage ID Tokens (OpenID Connect) and Access Tokens (OAuth 2.0).
 
+[More information about scopes](https://www.oauth.com/oauth2-servers/scope/) | 
+[More information about Access Tokens](https://www.oauth.com/oauth2-servers/access-tokens/)
+
 ### Custom Claims
 
 The JWT extension to the OAuth Framework lets you include custom claims in ID and Access Tokens.
```
Comment (richardmateosian-okta) [inline _source/_use_cases/api_security/index.md]: In the first picture, the first bullet should say ID Tokens (plural).

File: _source/_use_cases/api_security/index.md (hunk 9 above)
Comment (tthompson-okta) [inline _source/_use_cases/api_security/index.md]: Hmm, do you need to push the latest? It still says "authentication" as of now.

File: _source/_use_cases/api_security/index.md [hunk 15]
```diff
@@ -43,16 +44,93 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+
+[The Sign-in Widget](https://github.com/okta/okta-signin-widget) is Okta's open-source, embeddable widget for end-to-end sign-in flow with MFA and account recovery:
+
+* Returns an ID Token for an authenticated user 
+* Supports social authentication providers 
+* Is built on the [Okta Auth SDK](/code/javascript/okta_auth_sdk.html)
+
+### Okta Auth SDK
+
+[The Okta Auth SDK](https://github.com/okta/okta-auth-js) is also open source.
```
Comment (tthompson-okta) [inline _source/_use_cases/api_security/index.md]: Nit: the SDK's are all open-source, and we mentioned this at the opening of this section (on line 53). Sounds redundant here. I'd just combine this with the next sentence, i.e. "The Okta Auth SDK wraps authentication and session management APIs..."

File: _source/_use_cases/api_security/index.md [hunk 16]
```diff
@@ -43,16 +44,93 @@ Using Okta's OAuth-as-a-Service feature, API Access Management, provides many be
...
+
+Okta's Org Authorization Server requires no set up or configuration to mint ID Tokens and Access Tokens. 
+This authorization server supports OpenID Connect-only use cases.
+
+* Access policies are assignment based. Access is granted based on whether the user is assigned to the client app in Okta (directly or via groups).
+* Scopes in a request are either all granted, or none are. 
+* Audience (`aud`) for Access Tokens is always the Okta org:
+    * If an Access Token is requested, it's only good for Okta APIs. The token can't be used for custom or internal APIs.
+    * The Access Token format is opaque and shouldn't be parsed by the client. 
+* The Org Authorization Server is referenced via the root OAuth namespace: `/oauth2/v1/[authorize|token|introspect|revoke]`.
```
Comment (tthompson-okta) [inline _source/_use_cases/api_security/index.md]: I think we need to link to the reference for these APIs here, or otherwise elaborate on the difference between the "root namespace" versus the "authorization server namespace".

File: _source/_use_cases/api_security/index.md [hunk 17]
```diff
@@ -21,11 +27,6 @@ When do you use API Access Management and when do you use OpenID Connect?
 In general, use OpenID Connect to sign users into apps, and use API Access Management to secure your APIs:
```
Comment (karlmcguinness-okta) [inline _source/_use_cases/api_security/index.md]: Use OIDC to authenticate a user and create/manage sessions.  Use OAuth2 for delegated authorization to a API resource.  You don't need a session to access a API resource but often need a session to sign into an app

File: _source/_use_cases/api_security/index.md [hunk 18]
```diff
@@ -7,11 +7,17 @@ excerpt: Secure your APIs with Okta's implementation of the OAuth 2.0 standard.
 
 # Overview
 
-Use API Access Management, Okta's implementation of the OAuth 2.0 standard, to secure your APIs.
+Use API Access Management, Okta's implementation of the OAuth 2.0 protocol, to secure your APIs.
 API Access Management is integrated with Okta's implementation of OpenID Connect for authentication;
 OpenID Connect is also available separately. Similarly, Okta provides a client management API
```
Comment (karlmcguinness-okta) [inline _source/_use_cases/api_security/index.md]: This points seems somewhat muddled.  Okta provides an OpenID Connect certified implementation for authentication as well as OAuth 2.0 Authorization Server(s) that apps can use to implement delegated authorization with app-specific scopes and claims using industry standard OAuth flows.

Maybe too wordy for an overview, so trust your insights

File: _source/_use_cases/api_security/index.md (hunk 9 above)
Comment (karlmcguinness-okta) [inline _source/_use_cases/api_security/index.md]: Feels like we could use an analogy here to cement the conceptual model.  We have used valet keys or hotel key cards before to illustrate

File: _source/_use_cases/api_security/index.md (hunk 17 above)
Comment (karlmcguinness-okta) [inline _source/_use_cases/api_security/index.md]: +1

File: _source/_use_cases/api_security/index.md [hunk 19]
```diff
@@ -21,11 +27,6 @@ When do you use API Access Management and when do you use OpenID Connect?
 In general, use OpenID Connect to sign users into apps, and use API Access Management to secure your APIs: 
 create one or more authentication servers, define scopes and claims, and create policies and rules to determine who can access your API resources.
```
Comment (karlmcguinness-okta) [inline _source/_use_cases/api_security/index.md]: authorization

File: _source/_use_cases/api_security/index.md [hunk 20]
```diff
@@ -21,11 +27,6 @@ When do you use API Access Management and when do you use OpenID Connect?
...
 
-For example:
-
-* Use Case 1 (API Access Management): You need to control API access for a variety of consumers: vendors, employees, and customers, for example.
-* Use Case 2 (OpenID Connect): You want users to sign into your custom web application to access their account. 
-
 ### Complex Use Cases
 
 You can also specify authorization servers in your OpenID Connect API calls.
```
Comment (karlmcguinness-okta) [inline _source/_use_cases/api_security/index.md]: OpenID Connect or OAuth 2.0.  Not just restricted to OIDC

File: _source/_use_cases/api_security/index.md (hunk 13 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: @tthompson-okta Can you clarify if it's important to say "referenced" here? I had the sense it was.

File: _source/_use_cases/api_security/index.md (hunk 12 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: @richardmateosian-okta I was trying to unpack four-noun noun phrases (see how hard they are to read?). Is there something better that doesn't repack that suitcase?

File: _source/_use_cases/api_security/index.md (hunk 18 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: @tthompson-okta Can you weigh in here? I like Karl's paragraph, but it seems to contradict line 12 where we point out OIDC is available separately. I want to do justice to the feature but not confuse customers re: the licensing/SKU.

File: _source/_use_cases/api_security/index.md (hunk 12 above)
Comment (richardmateosian-okta) [inline _source/_use_cases/api_security/index.md]: This authorization server supports using OpenID Connect [by itself, without xxx].  
This authorization server supports only OpenID Connect [not other protocols].

It depends on what the sentence is trying to convey.

File: _source/_use_cases/api_security/index.md (hunk 18 above)
Comment (cwu-okta) [inline _source/_use_cases/api_security/index.md]: I don't think Karl's suggestion contradicts line 12. OIDC and OAuth are separate products and do not necessarily need to be used together. 

I agree with you on not wanting to confuse customers on how to purchase the product. Karl's suggestion makes it clear the two are different products and doesn't mention anything about how to purchase/make available these features.

File: _source/_use_cases/api_security/index.md (hunk 6 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Added link.

File: _source/_use_cases/api_security/index.md (hunk 9 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: @karlmcguinness-okta I added a very light version, is this enough?

[OAuth 2.0](https://www.oauth.com) is an open standard that defines how to perform authentication securely. It decouples authentication policy decisions from enforcement; like hotel key cards, but for apps. API Access Management is Okta's implementation of an authorization service which leverages Okta's Universal Directory, groups, policies, multifactor authentication (MFA) and other features.

File: _source/_use_cases/api_security/index.md (hunk 12 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Perfect. I went with this: This authorization server supports use cases that need OpenID Connect but not OAuth 2.0.

File: _source/_use_cases/api_security/index.md (hunk 13 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: we don't mention "root OAuth namespace" in the existing doc, so changing to "API requests include the Org Authorization Server: /oauth2/:authorizationServerId/v1[authorize...."

File: _source/_use_cases/api_security/index.md (hunk 14 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: done.

File: _source/_use_cases/api_security/index.md (hunk 15 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

File: _source/_use_cases/api_security/index.md (hunk 16 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: We never mention the OAuth namespace in the doc, I changed it to this (and will add link):

* API requests include the Org Authorization Server: `/oauth2/:authorizationServerId/v1/[authorize|token|introspect|revoke]`.

File: _source/_use_cases/api_security/index.md (hunk 17 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

File: _source/_use_cases/api_security/index.md (hunk 18 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: Done.

File: _source/_use_cases/api_security/index.md (hunk 19 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: done.

File: _source/_use_cases/api_security/index.md (hunk 20 above)
Comment (mystiberry-okta) [inline _source/_use_cases/api_security/index.md]: done.
